src/protocol/malicious.rs

```diff
@@ -158,16 +189,19 @@ impl<F: Field> SecurityValidator<F> {
     /// ## Panics
     /// Will panic if the mutex is poisoned
     #[allow(clippy::await_holding_lock)]
-    pub async fn validate(self, ctx: SemiHonestContext<'_, F>) -> Result<(), Error> {
+    pub async fn validate<D: DowngradeMalicious>(self, values: D) -> Result<D::Target, Error> {
         // send our `u_i+1` value to the helper on the right
-        let channel = ctx.mesh();
-        let helper_right = ctx.role().peer(Direction::Right);
-        let helper_left = ctx.role().peer(Direction::Left);
+        let channel = self.propagate_uw_ctx.mesh();
+        let helper_right = self.propagate_uw_ctx.role().peer(Direction::Right);
+        let helper_left = self.propagate_uw_ctx.role().peer(Direction::Left);
 
-        let state = self.u_and_w.lock().unwrap();
+        let (u_local, w_local) = {
+            let state = self.u_and_w.lock().unwrap();
+            (state.u, state.w)
+        };
```

Comment:
What is the benefit of doing this?

Reply:
This limits the scope of the lock. That turned out to be a big problem building the malicious runner for tests.

Reply:
Note also that when you overrode the clippy warning, you were burying a problem for later.

Reply:
I should explain more, but the mammoth warning I got when trying to build out something that used this was amazing.  The only clue I got was that it was complaining about `MutexGuard<AccumulatorState<_>>` not implementing `Send`.